test/test_torch.py

```diff
@@ -4341,6 +4341,90 @@ def test_diff(self, device, dtype):
             scalar = torch.tensor(2, device=device, dtype=dtype)
             torch.diff(scalar)
 
+    @onlyOnCPUAndCUDA
+    @dtypes(torch.long, torch.float32, torch.complex64)
+    def test_gradient_tensor_list(self, device, dtype):
+        inputs = (
+            ((4, 4, 4), [(1., 3.0, 6., 10.), (1., 3.0, 4., 90.), (1., 2.5, 3.4, 7.)], [0, 1, 2], 1),
+            ((4, 4, 4, 3), [(1., 3.0, 6., 10.), (1., 2.5, 7.)], [0, -1], 1),
+            ((4, 5, 2, 4, 3), [(1., 3.0, 6., 10.), (1., 3.4, 7.)], [-2, -1], 1),
+            ((4, 3, 6, 5, 3), [(1., 3.0, 6., 10.), (2.5, 3.4, 7.)], [0, 1], 1),
+            ((4, 3, 2, 5, 3), [(1., 3.0), (3.4, 7., 6.0)], [2, 1], 1),
+            ((4, 4, 4, 3), [1.23, (1., 2.5, 3.4,)], [0, -1], 1),
+            ((4, 5, 2, 4, 3), [(1., 3.0, 6., 10.), 3.45], [-2, -1], 1),
+        )
+
+        for size, spacing, axis, edge_order in inputs:
+            for contig in [True, False]:
+                t = make_tensor(size, device, dtype, noncontiguous=not contig)
+                t_n = t.cpu().numpy()
+                spacing_list = []
+                for space in spacing:
+                    spacing_list.append(torch.tensor(space, dtype=float, device=device))
+                result = torch.gradient(t, spacing=spacing_list, dim=axis, edge_order=edge_order)
+                result_n = np.gradient(t_n, *spacing, axis=axis, edge_order=edge_order)
+                if not isinstance(result_n, list):
```

Comment:
This is because torch.gradient() will always return a list, but when returning a single tensor NumPy will return just the tensor?

Reply:
yes exactly :)